### PR TITLE
[Main] Backup and Restore Extensions

### PR DESCRIPTION
**Pull request by nickwsuse**
Extensions in this PR are to enable and enhance backup/restore checks.

### CODE REVIEW

**Review by caliskanugur — DISMISSED** [review]

**Review by igomez06 — DISMISSED** [review]

File: extensions/charts/backuprestore.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	backupChartInstallActionPayload := &payloadOpts{
+		InstallOptions: *installOptions,
+		Name:           backupChartName,
+		Namespace:      backupChartNamespace,
+		Host:           serverSetting.Value,
+	}
+
+	chartInstallAction := newBackupChartInstallAction(backupChartInstallActionPayload, false, rancherBackupOpts)
+
+	if withStorage {
```
Comment (joesims22) [inline extensions/charts/backuprestore.go:48]: nit, could this if statement be revised to an if/else statement so `chartInstallAction` is only assigned once depending on the value of ```withStorage```?

**Review by joesims22 — APPROVED** [review]
left one nit comment, otherwise lgtm

File: extensions/charts/backuprestore.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		Name:           backupChartName,
+		Namespace:      backupChartNamespace,
+		Host:           serverSetting.Value,
+	}
+
+	if withStorage {
+		chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, true, rancherBackupOpts)
+	} else {
+		chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, false, rancherBackupOpts)
+	}
```
Comment (Priyashetty17) [inline extensions/charts/backuprestore.go:50]: Can you replace this if/else statement with: 
`chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, withStorage, rancherBackupOpts)`

**Review by Priyashetty17 — COMMENTED** [review]
Are the changes to go.mod and go.sum intentional?

**Comment by nickwsuse** [comment]
> Are the changes to go.mod and go.sum intentional?

I only ran go mod tidy after making my changes, so anything else is not intentional. If I don't include the go.mod and go.sum files though the checks have failed in the past, so I assumed they were required to actually merge.

**Review by Priyashetty17 — APPROVED** [review]
lgtm

**Comment by anupama2501** [comment]
I think these should be all moved to actions based on our previous discussions about actions vs extensions.. since we do have code freeze on shepherd. can we hold off on this and then move this to actions and close this PR?

**Comment by igomez06** [comment]
> I think these should be all moved to actions based on our previous discussions about actions vs extensions.. since we do have code freeze on shepherd. can we hold off on this and then move this to actions and close this PR?

Yes agreed, Thanks Anu

**Comment by nickwsuse** [comment]
These have now been moved into the actions section of r/r per these PRs:
https://github.com/rancher/rancher/pull/47063/files
https://github.com/rancher/rancher/pull/45057

Closing this PR since it's no longer needed.

**Pull request closed without merging**